egui/src/widgets/plot/legend.rs

```diff
@@ -1,81 +1,231 @@
-    pub text: String,
-    pub color: Color32,
-    pub checked: bool,
-    pub hovered: bool,
+use super::items::PlotItem;
+
+/// Where to place the plot legend.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum LegendPosition {
+    TopLeft,
+    TopRight,
+    BottomLeft,
+    BottomRight,
+}
+
+impl LegendPosition {
+    pub fn all() -> impl Iterator<Item = LegendPosition> {
+        [
+            LegendPosition::TopLeft,
+            LegendPosition::TopRight,
+            LegendPosition::BottomLeft,
+            LegendPosition::BottomRight,
+        ]
+        .iter()
+        .copied()
+    }
+}
+
+/// The configuration for a plot legend.
+#[derive(Clone, Copy, PartialEq)]
+pub struct Legend {
+    pub text_style: TextStyle,
+    pub position: LegendPosition,
+}
+
+impl Default for Legend {
+    fn default() -> Self {
+        Self {
+            text_style: TextStyle::Body,
+            position: LegendPosition::TopRight,
+        }
+    }
+}
+
+#[derive(Clone)]
+struct LegendEntry {
+    color: Color32,
+    checked: bool,
+    hovered: bool,
 }
 
 impl LegendEntry {
-    pub fn new(text: String, color: Color32, checked: bool) -> Self {
+    fn new(color: Color32, checked: bool) -> Self {
         Self {
-            text,
             color,
             checked,
             hovered: false,
         }
     }
 }
 
-impl Widget for &mut LegendEntry {
+impl Widget for (&String, &mut LegendEntry) {
     fn ui(self, ui: &mut Ui) -> Response {
-        let LegendEntry {
-            checked,
+        let (
             text,
-            color,
-            ..
-        } = self;
-        let icon_width = ui.spacing().icon_width;
-        let icon_spacing = ui.spacing().icon_spacing;
-        let padding = vec2(2.0, 2.0);
-        let total_extra = padding + vec2(icon_width + icon_spacing, 0.0) + padding;
-
-        let text_style = TextStyle::Button;
-        let galley = ui.fonts().layout_no_wrap(text_style, text.clone());
-
-        let mut desired_size = total_extra + galley.size;
-        desired_size = desired_size.at_least(ui.spacing().interact_size);
-        desired_size.y = desired_size.y.at_least(icon_width);
-
+            LegendEntry {
+                color,
+                checked,
+                hovered,
+            },
+        ) = self;
+
+        let galley = ui
+            .fonts()
+            .layout_no_wrap(ui.style().body_text_style, text.clone());
+
+        let icon_size = galley.size.y;
+        let icon_spacing = icon_size / 5.0;
+        let total_extra = vec2(icon_size + icon_spacing, 0.0);
+
+        let desired_size = total_extra + galley.size;
         let (rect, response) = ui.allocate_exact_size(desired_size, Sense::click());
-        let rect = rect.shrink2(padding);
 
         response.widget_info(|| WidgetInfo::selected(WidgetType::Checkbox, *checked, &galley.text));
 
         let visuals = ui.style().interact(&response);
+        let flipped = ui.layout().cross_align() == Align::RIGHT;
```

Comment:
flipped vs what? Better to call it something like `label_on_the_left` (or is it `label_on_the_right`?)

Reply:
:+1: Changed